caesars-palace/blocks/slider-card/slider-card.js

```diff
@@ -1,42 +1,75 @@
 export default function decorate(block) {
   const cardWrapper = document.createElement('div');
   cardWrapper.classList.add('card-wrapper');
-  block.querySelectorAll('div.slider-card > div').forEach((div) => {
+
+  function isADesktop() {
+    const mediaDesktop = window.matchMedia('only screen and (min-width: 769px)');
+    return mediaDesktop.matches;
+  }
+
+  // default value is 3
+  let numberOfCardsDisplayed = 3;
+  block.querySelectorAll('div.slider-card > div').forEach((div, idx) => {
+    // We do not need to add a class to the first div as it includes the number of
+    // cards we wish to display
+    if (idx === 0) {
+      numberOfCardsDisplayed = div.children[0].innerHTML;
+      div.style.display = 'none';
+      if (isADesktop()) {
+        const sliderCard = block;
+        sliderCard.style = `width: calc((100%/${numberOfCardsDisplayed} - 10px)); gap: 10px;`;
+      }
+      return;
+    }
     div.classList.add('card');
-    const imageDiv = div.children[0];
-    imageDiv.children[0].children[3].classList.add('card-image');
-    div.children[1].classList.add('short-description');
-    div.children[2].classList.add('long-description');
+    let index = 0;
+    if (div.getElementsByTagName('picture').length > 0) {
+      const imageDiv = div.children[index];
+      imageDiv.children[index].children[3].classList.add('card-image');
+      index += 1;
+    }
+    div.children[index].classList.add('short-description');
+    div.children[index + 1].classList.add('long-description');
```

Comment:
I think this logic could be improved to rely less on indexes which may be more error prone. You could locate the picture via `div.querySelector('picture')`. If the picure exists you could then find its parent div by `picture.closest('div')`. To get all the other divs you could add a class to the image div and then could query for all other sibling divs.

eg.
```
const contentDivs = div.querySelectorAll(':scope > div:not(.card-image)');
```

Reply:
+1